tests/integration/decorators-test.js

```diff
@@ -212,4 +213,159 @@ module('Integration | decorators', function(hooks) {
     assert.dom('.test-component').doesNotHaveClass('foo');
     assert.dom('.test-component').doesNotHaveClass('bar');
   });
+
+  test('it can decorate a simple ES5 getter', async function(assert) {
+    let hbs = setupStyles({
+      'simple-bool-class': 'simple-bool--true',
+
+      'bool-true': 'explicit-bool--true',
+      'bool-false': 'explicit-bool--false',
+
+      'foo': 'string--foo',
+      'bar': 'string--bar'
+    });
+
+    this.setProperties({
+      simpleBool: false,
+      explicitBool: false,
+      string: null
+    });
+
+    @layout(hbs``)
+    @classNames('test-component')
+    class TestComponent extends Component {
+      simpleBool;
+      explicitBool;
+      string;
+
+      @localClassName
+      get simpleBoolClass() {
+        return this.simpleBool;
+      }
+
+      @localClassName('bool-true', 'bool-false')
+      get explicitBoolClass() {
+        return this.explicitBool;
+      }
+
+      @localClassName
+      get stringClass() {
+        return this.string;
+      }
+
+      didReceiveAttrs() {
+        // Should be `super.didReceiveAttrs()`, however there is an
+        // incompatibility between the Babel class transpilation and the Ember
+        // Object Model.
+        this._super();
```

Comment:
Is there? I thought `Component` just didn't have a base `didReceiveAttrs` implementation (so there's no `super` to call)?